core/ledger/kvledger/txmgmt/privacyenabledstate/common_storage_db.go

```diff
@@ -335,6 +336,52 @@ func (s *CommonStorageDB) HandleChaincodeDeploy(chaincodeDefinition *cceventmgmt
 	return nil
 }
 
+// HandleInProcChaincodeDeploy adds DB indexes for the chaincode
+func (s *CommonStorageDB) HandleInProcChaincodeDeploy(chaincodeDefinition *cceventmgmt.ChaincodeDefinition, artifacts map[string]*api.DBArtifacts) error {
+	//Check to see if the interface for IndexCapable is implemented
+	indexCapable, ok := s.VersionedDB.(statedb.IndexCapable)
+	if !ok {
+		return nil
+	}
+
+	if chaincodeDefinition == nil {
+		return errors.New("chaincode definition not found while creating couchdb index")
+	}
+
+	dbArtifacts, ok := artifacts[indexCapable.GetDBType()]
+	if !ok {
+		logger.Debugf("No index definitions for DB of type [%s] for chaincode [%s]", indexCapable.GetDBType(), chaincodeDefinition.Name)
+		return nil
+	}
+
+	collectionConfigMap, err := extractCollectionNames(chaincodeDefinition)
+	if err != nil {
+		logger.Errorf("Error while retrieving collection config for chaincode=[%s]: %s",
+			chaincodeDefinition.Name, err)
+		return nil
+	}
+
+	logger.Debugf("Adding %d index(es) for chaincode [%s]", len(dbArtifacts.Indexes), chaincodeDefinition.Name)
+	if err := indexCapable.ProcessIndexes(chaincodeDefinition.Name, dbArtifacts.Indexes); err != nil {
+		logger.Errorf("Error processing index for chaincode [%s]: %s", chaincodeDefinition.Name, err)
```

Comment:
no return here, is this OK?

Reply:
Same here - I copied the existing logic but I don't think it's right. I'll change it to return an error.